lib/smooth/application/io/i2c/SHT30.cpp

```diff
@@ -228,7 +234,59 @@ namespace smooth::application::sensor
 
         if (!res)
         {
-            Log::error(TAG, "Execute read block failed");
+            Log::error(TAG, "Execute read_block failed");
+        }
+
+        return res;
+    }
+
+    // Execute read16 that uses slave clock-stretching to read data from SHT30
+    bool SHT30::execute_read16(FixedBufferBase<uint8_t>& raw_data)
+    {
+        // Read the measurement data using slave clock-stretching.  The master_timeout set to 560000 = 7ms,
+        // this timeout will satisfy both clock-stretching modes
+        bool res = read16(address, repeatability_command[repeatability_mode], raw_data, true, true, scl_timeout_7ms);
+
+        if (!res)
+        {
+            Log::error(TAG, "Execute read16 failed");
+        }
+
+        return res;
+    }
+
+    // Execute measurement read
+    bool SHT30::execute_measurement_read(FixedBufferBase<uint8_t>& raw_data)
+    {
+        bool res = false;
+
+        if (use_clock_stretching)
+        {
+            res = execute_read16(raw_data);
+        }
+        else
+        {
+            res = execute_repeatability_command();
+            res &= execute_read_block(raw_data);
```

Comment:
I know I'm guilty of using this construct too, but I've  since then learned that &= does *not* evaluate using short circuit-logic so please change it as follows.

```suggestion
            res = execute_repeatability_command() && execute_read_block(raw_data);
```